Doc/library/locale.rst

```diff
@@ -365,12 +365,26 @@ The :mod:`locale` module defines the following exception and functions:
    Please note that this function will only work for exactly one %char specifier.
    For whole format strings, use :func:`format_string`.
 
+   .. deprecated:: 3.7
+      Use :meth:`format_string` instead
```

Comment:
So looking at this again, I think what we should do is move 'format' under 'format_string', and change *its* body to say "works like format_string but only accepts a single format specification.  That followed by the deprecation notice will make it a lot less likely anyone will use it :)